03week/ticTacToe.js

```diff
@@ -23,26 +25,112 @@ function printBoard() {
   console.log('2 ' + board[2].join(' | '));
 }
 
+
+
 function horizontalWin() {
+  if (board.includes("X", [0, 0], [0, 1], [0, 2])) {
+    return true;
+  } else if (board.includes("X", [1, 0], [1, 1], [1, 2])) {
+    return true;
+  } else if (board.includes("X", [2, 0], [2, 1], [2, 2])) {
+    return true;
+  }
   // Your code here
+  //check each array for  3 X or O
+  //check 1st array 0 and all elements for x, y. and all positions are returned, then x or y wins
+  //check 2nd array 1 and all elements for x, y. and all positions are returned, then x or y wins
+  //check 3rd array 2 and all elements for x, y. and all positions are returned, then x or y wins
+
+
+  // winner combos is array 0 in 0 1 2 || array 1 in 0, 1, 2 || in array 2 in 0, 1, 2
+  // console.log(board[0][0])
+
 }
 
 function verticalWin() {
+  if (board.includes("X", [0, 0], [1, 0], [2, 0])) {
+    return true;
+  } else if (board.includes("X", [0, 1], [1, 1], [2, 1])) {
+    return true;
+  } else if (board.includes("X", [0, 2], [1, 2], [2, 2])) {
+    return true;
+  }
+
+
   // Your code here
+  //check each element 0, 1, 2 in each array for 3 x or 3 O
+  //check element 0 in array 0 for x, y
+  //check element 0 in array 1 for x,y
+  //check element 0 in array 2 for xy
+  //then x,y wins
+  //check element 1 in array 0 for x, y
+  //check element 1 in array 1 for x,y
+  //check element 1 in array 2 for xy
+  //then x,y wins
+  //check element 2 in array 0 for x, y
+  //check element 2 in array 1 for x,y
+  //check element 2 in array 2 for xy
+  //then x,y wins
+
+  // winner combos in array 0 index 0 + in array 1 index 0 + array 2 index 0 || array 0 index 1 + array 1  index 1 + array 2 index 1 || array 0 index 2 + array 1 index 2 + array 2 index 2
+
 }
 
 function diagonalWin() {
+  if (board.includes("X", [0, 0], [1, 1], [2, 2])) {
+    return true;
+  } else if (board.includes("X", [2, 0], [1, 1], [0, 2])) {
+    return true;
+  }
+
+
   // Your code here
+  //check element 0 in array 0
+  //check element 1 in array 1,
+  //check element 2 in array 2
+  //then x,y wins
 }
 
 function checkForWin() {
-  // Your code here
+  if (horizontalWin(), true) {
```

Comment:
try using the || operator here so you don't have to keep rewriting the same code. 
Ex: if(horizontalWin() || .....) {do something}